### PR TITLE
Optimize starters auto-configuration. (main)

### PR DESCRIPTION
**Pull request by misselvexu**
## PR Type
Refactoring.

## Describe what this PR does for and how you did.
Optimize SCT starters auto-configuration

## Adding the issue link (#xxx) if possible.
None.

## Note
None.

## Checklist

- [x] Add copyright holder at the beginning of .class file if it is new.
- [x] Add information of this PR to CHANGELOG.md in root of project.
- [x] All junit tests passing.
- [ ] Coverage from `Codecov Report` should not decrease.

## Checklist (Optional)

- [x] Will Pull Request to branch of 2020.0 and 2021.0.
- [x] Add documentation in javadoc in code or comment below the PR if necessary.
- [x] Add your name as @author to the beginning of .class file.

### CODE REVIEW

File: spring-cloud-starter-tencent-polaris-circuitbreaker/src/main/java/com/tencent/cloud/polaris/circuitbreaker/config/PolarisCircuitBreakerAutoConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	 *
+	 * @author wh 2022/6/21
+	 */
+	@Configuration(proxyBeanMethods = false)
+	@AutoConfigureAfter(PolarisContextAutoConfiguration.class)
+	@ConditionalOnProperty(value = "spring.cloud.polaris.circuitbreaker.enabled", havingValue = "true", matchIfMissing = true)
+	protected static class PolarisRestTemplateAutoConfiguration {
+
+		@Bean
+		@ConditionalOnBean(RestTemplate.class)
```
Comment (SkyeBeFreeman) [inline spring-cloud-starter-tencent-polaris-circuitbreaker/src/main/java/com/tencent/cloud/polaris/circuitbreaker/config/PolarisCircuitBreakerAutoConfiguration.java:89]: `@ConditionalOnBean(RestTemplate.class)` 放在上层是不是就行了？

File: spring-cloud-starter-tencent-metadata-transfer/src/main/java/com/tencent/cloud/metadata/core/EncodeTransferMedataScgFilter.java [hunk 2]
```diff
@@ -69,7 +69,7 @@ public Mono<Void> filter(ServerWebExchange exchange, GatewayFilterChain chain) {
 			String metadataStr = JacksonUtils.serialize2Json(customMetadata);
 			try {
 				builder.header(MetadataConstant.HeaderName.CUSTOM_METADATA,
-						URLEncoder.encode(metadataStr, StandardCharsets.UTF_8.name()));
```
Comment (SkyeBeFreeman) [inline spring-cloud-starter-tencent-metadata-transfer/src/main/java/com/tencent/cloud/metadata/core/EncodeTransferMedataScgFilter.java:72]: 为啥不用这个？

Comment (misselvexu) [reply spring-cloud-starter-tencent-metadata-transfer/src/main/java/com/tencent/cloud/metadata/core/EncodeTransferMedataScgFilter.java:72]: Just seems that there are too many global references to character codes. ~ 😂

File: spring-cloud-starter-tencent-polaris-circuitbreaker/src/main/java/com/tencent/cloud/polaris/circuitbreaker/config/PolarisCircuitBreakerAutoConfiguration.java (hunk 1 above)
Comment (misselvexu) [reply spring-cloud-starter-tencent-polaris-circuitbreaker/src/main/java/com/tencent/cloud/polaris/circuitbreaker/config/PolarisCircuitBreakerAutoConfiguration.java:89]: > `@ConditionalOnBean(RestTemplate.class)` 放在上层是不是就行了？

Good idea .

File: spring-cloud-starter-tencent-polaris-circuitbreaker/src/main/java/com/tencent/cloud/polaris/circuitbreaker/config/PolarisCircuitBreakerAutoConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+	/**
+	 * Configuration for Polaris {@link RestTemplate} which can automatically bring in the call
+	 * results for reporting.
+	 *
+	 * @author wh 2022/6/21
+	 */
+	@Configuration(proxyBeanMethods = false)
+	@AutoConfigureAfter(PolarisContextAutoConfiguration.class)
+	@ConditionalOnBean(RestTemplate.class)
```
Comment (SkyeBeFreeman) [inline spring-cloud-starter-tencent-polaris-circuitbreaker/src/main/java/com/tencent/cloud/polaris/circuitbreaker/config/PolarisCircuitBreakerAutoConfiguration.java:85]: 使用 `@ConditionalOnClass` 是不是好点？否则将受限于bean加载顺序

Comment (misselvexu) [reply spring-cloud-starter-tencent-polaris-circuitbreaker/src/main/java/com/tencent/cloud/polaris/circuitbreaker/config/PolarisCircuitBreakerAutoConfiguration.java]: done.

**Review by SkyeBeFreeman — APPROVED** [review]
LGTM.

**Pull request merged**